lib/SILGen/SILGenPoly.cpp

```diff
@@ -7043,6 +7043,14 @@ void SILGenFunction::emitProtocolWitness(
   FullExpr scope(Cleanups, cleanupLoc);
   FormalEvaluationScope formalEvalScope(*this);
 
+  // The protocol conditional conformance itself might bring some T :
+  // Differentiable conformances. They are already added to the derivative
+  // generic signature. Update witness substitution map generic signature to
+  // have them as well.
+  if (auto *derivativeId = witness.getDerivativeFunctionIdentifier())
+    witnessSubs = SubstitutionMap::get(derivativeId->getDerivativeGenericSignature(),
```

Comment:
Should the correct witnessSubs be computed in Sema? See lib/AST/RequirementEnvironment.cpp

Reply:
It seems to be calculated correctly there:
`<τ_0_0 where τ_0_0 : Differentiable>` is a witness generic thunk signature and the corresponding requirement-to-witness substitution map is:
```
(substitution_map generic_signature=<Self where Self : P>
  (substitution Self ->
    (bound_generic_struct_type decl="main.(file).Wrapper@75711.swift:3:8"
      (generic_type_param_type depth=0 index=0 param_kind=type)))
  (conformance type="Self"
    (normal_conformance type="Wrapper<T>" protocol="P"
      (assoc_type req="T" type="T")
      (assoc_conformance type="Self" proto="Copyable"
        (builtin_conformance type="Wrapper<T>" protocol="Copyable"))
      (assoc_conformance type="Self" proto="Escapable"
        (builtin_conformance type="Wrapper<T>" protocol="Escapable"))
      (assoc_conformance type="Self.T" proto="Differentiable"
        (abstract_conformance protocol="Differentiable"))
      (requirement "T" conforms_to "Differentiable"))))
```

However, `witnessSubs` here is:
```
(substitution_map generic_signature=<T where T : Copyable, T : Escapable>
  (substitution T ->
    (primary_archetype_type address=0x15c9d8c18 conforms_to="_Differentiation.(file).Differentiable" name="\xCF\x84_0_0"
      (interface_type=generic_type_param_type depth=0 index=0 param_kind=type)))
  (conformance type="T"
    (abstract_conformance protocol="Copyable"))
  (conformance type="T"
    (abstract_conformance protocol="Escapable")))
```

Reply:
@slavapestov So, what do you think about this? Should we change the calculation somehow? For now I just made it conditional for autodiff and only adding `T : Differentiable` requirement. W/o this we are unable to calculate derivative function type in protocol conformance as we need to have corresponding associated types etc.

Reply:
@slavapestov Any further suggestions here?